test/generators/csharp/presets/__snapshots__/JsonSerializerPreset.spec.ts.snap

```diff
@@ -85,7 +85,7 @@ internal class TestConverter : JsonConverter<Test>
         }
       if (propertyName == \\"enumProp\\")
         {
-          var value = EnumTestExtensions.ToEnumTest(JsonSerializer.Deserialize<dynamic>(ref reader, options));
+          var value = EnumTestExtensions.ToEnumTest(JsonSerializer.Deserialize<string>(ref reader, options));
```

Comment:
I think I get what you're asking but to avoid confusion, can you kindly post an example schema? Thanks :)

Reply:
```
{  "enum": ["string", 2]}
```

It's actually part of the current enum, that can both be boolean, string, number and JSON encoded string: https://github.com/asyncapi/modelina/blob/4bf0b566465c99127da0befc95a3ba1a31a54e6d/test/generators/csharp/presets/__snapshots__/JsonSerializerPreset.spec.ts.snap#L177

Reply:
Happy to accept a PR that partially fix it, as long as it is improved compared to the current approach, and we create a followup issue with the restriction 🙂

Reply:
I've dug into this a little bit and the code we would end up generating is a little bit ugly.

Crucially, `System.Text.Json` will return a `dynamic` value as a `JsonElement`, so we would need to look for JsonElement's and handle all the possible scenarios. The code ends up looking like:

```c#
if (JsonSerializer.Deserialize<dynamic>(ref reader, options) is JsonElement jsonElement)
{
    if (jsonElement.ValueKind == JsonValueKind.Null)
    {
        instance.EnumProp = null;
        continue;
    }
    else if (jsonElement.ValueKind == JsonValueKind.String)
    {
        instance.EnumProp = DynamicEnumExtensions.ToDynamicEnum(jsonElement.GetString());
        continue;
    }
    else if (jsonElement.ValueKind == JsonValueKind.Number)
    {
        instance.EnumProp = DynamicEnumExtensions.ToDynamicEnum(jsonElement.GetInt32());
        continue;
    }
    else if (jsonElement.ValueKind == JsonValueKind.Object ||
                jsonElement.ValueKind == JsonValueKind.Array)
    {
        instance.EnumProp = DynamicEnumExtensions.ToDynamicEnum(jsonElement.ToString());
        continue;
    }
    else if (jsonElement.ValueKind == JsonValueKind.True || jsonElement.ValueKind == JsonValueKind.False)
    {
        instance.EnumProp = DynamicEnumExtensions.ToDynamicEnum(jsonElement.GetBoolean());
        continue;
    }
    else
    {
        continue;
    }
}
else
{
    var value2 = EnumTestExtensions.ToEnumTest(JsonSerializer.Deserialize<string>(ref reader, options));
    instance.EnumTest = value2;
    continue;
}
```

Let me know if:

1. You want me to implement the above,
2. You are happy to merge the code as-is, or
3. You can think of a better way 😄 

I'm happy with any of the above options, so just let me know 🙂.

Reply:
Sorry, yes - my posted code is pseudo-code as well. The key point I was trying to highlight is that we will need to inspect `JsonElement`, and so will have that big `if (jsonElement.TokenType== xxx)` block.

In practice, I don't _think_ it will ever fall through to the `else` - I was planning to try out a few scenarios as part of finishing the implementation though to be sure. 

> Since EnumTestExtensions.ToEnumTest takes a dynamic type, is it not possible to mix the two? 
Yes, we can do that 🙂 

So, just to confirm we want to do option 1 (implement the code), right?

Reply:
Ahh okay, yea, go for it 👍 If the current code with `var value = EnumTestExtensions.ToEnumTest(JsonSerializer.Deserialize<dynamic>(ref reader, options));` does not work, this is definitely better 👍